File: CPP/Geometric Progression/GeometricProgression.cpp
#include <iostream>
#include <cmath>

double geometricProgression(double firstTerm, double commonRatio, int numberOfTerms)
{
    return firstTerm * std::pow(commonRatio, numberOfTerms - 1);
}

int main()
{
    double firstTerm = 0;
    double commonRatio = 0;
    int numberOfTerms = 0;

    std::cout << "First term: ";
    std::cin >> firstTerm;

    std::cout << "Common ratio: ";
    std::cin >> commonRatio;

    std::cout << "Number of terms: ";
    std::cin >> numberOfTerms;

    for(int termIndex = 0; termIndex < numberOfTerms; termIndex++)
    {
        std::cout << geometricProgression(firstTerm, commonRatio, termIndex + 1) << std::endl;
    }

    return 0;
}

File: Java/Geometric Progression/GeometricProgression.java
public class GeometricProgression {

    public static void main(String[] args) {

        System.out.println("Geometric Progression");
        System.out.println();

        double firstTerm = 7;
        double commonRatio = 2;
        int numberOfTerms = 9;

        System.out.println("first term: " + firstTerm);
        System.out.println("common ratio: " + commonRatio);
        System.out.println("number of terms: " + numberOfTerms);
        System.out.println();

        //Geometric Progression
        System.out.println("Geometric Progression: ");
        for (int i = 0; i < numberOfTerms; i++) {
            double ithTerm = firstTerm * Math.pow(commonRatio, i);
            System.out.println(ithTerm);
        }

    }

}